lib/src/editor/ui/editor.dart

```diff
@@ -101,6 +92,89 @@ class Editor extends StatelessWidget {
   }
 }
 
+class _Canvas extends StatelessWidget {
+  Widget? component;
+  TransformationController _transformation = TransformationController();
+  _Canvas(this.component);
+  @override
+  Widget build(BuildContext context) {
+    return Consumer<ZoomProvider>(
+      builder: (context, model, child) {
+        _transformation.value = Matrix4.identity()..scale(model.zoom);
+        return context.watch<DevicePreviewProvider>().show
+            ? _DevicePreviewCanvas(component, _transformation)
+            : _InteractiveViewerCanvas(component, _transformation);
+      },
+    );
+  }
+}
+
+class _DevicePreviewCanvas extends StatelessWidget {
+  final Widget? component;
+  final TransformationController controller;
+  _DevicePreviewCanvas(this.component, this.controller);
+  @override
+  Widget build(BuildContext context) {
+    return DevicePreview(
+      builder: (context) {
+        return InteractiveViewer(
+          panEnabled: true,
+          boundaryMargin: EdgeInsets.all(double.infinity),
+          child: component ?? const SizedBox.shrink(),
+          transformationController: controller,
+        );
+      },
+    );
+  }
+}
+
+class _Doc extends StatelessWidget {
+  List<ComponentState> states;
+  ComponentState? currentState;
+  _Doc(this.states, this.currentState);
+  @override
+  Widget build(BuildContext context) {
+    return SingleChildScrollView(
+      child: Center(
+        child: Container(
```

Comment:
`Container` here might be an unnecessary widget for the layout since you're not setting any properties on it.